### PR TITLE
add event unsend & videoPlayComplete

### PR DESCRIPTION
**Pull request by PamornT**
add new event  unsend and videoPlayComplete

### CODE REVIEW

File: src/Line/Bot/Webhook/Events.hs [hunk 1]
```diff
@@ -109,6 +109,13 @@ data Event =
                       , timestamp  :: EpochMilli
                       , things     :: Things
                       }
+  | UnSend            { source    :: Source
```
Comment (moleike) [inline src/Line/Bot/Webhook/Events.hs:112]:
```suggestion
  | EventUnsend            { source    :: Source
```

File: src/Line/Bot/Webhook/Events.hs [hunk 2]
```diff
@@ -109,6 +109,13 @@ data Event =
...
+                      , timestamp :: EpochMilli
+                      }
+  | VideoPlayComplete { replyToken :: ReplyToken
```
Comment (moleike) [inline src/Line/Bot/Webhook/Events.hs:115]:
```suggestion
  | EventVideoPlayComplete { replyToken :: ReplyToken
```

Comment (moleike) [reply src/Line/Bot/Webhook/Events.hs:115]: There is an additional field here, `videoPlayComplete.trackingId`.

File: src/Line/Bot/Webhook/Events.hs (hunk 1 above)
Comment (moleike) [reply src/Line/Bot/Webhook/Events.hs:112]: There is also an additional field we'd need to add, `unsend.messageId`

**Review by moleike — CHANGES_REQUESTED** [review]
Thanks for the patch 🙌
I left a couple of comments: we need to add some additional fields to make the event model consistent with the docs. Also, I renamed the constructors, so that they can be decoded correctly (look at the `FromJSON` instance for `Event`).

File: src/Line/Bot/Webhook/Events.hs [hunk 3]
```diff
@@ -109,6 +110,14 @@ data Event =
                       , timestamp  :: EpochMilli
                       , things     :: Things
                       }
+  | EventUnSend       { source    :: Source
```
Comment (moleike) [inline src/Line/Bot/Webhook/Events.hs:113]:
```suggestion
  | EventUnsend       { source    :: Source
```

File: src/Line/Bot/Webhook/Events.hs [hunk 4]
```diff
@@ -318,3 +327,13 @@ instance FromJSON Things where
     deviceId  <- o .: "deviceId"
     eventType <- o .: "type"
     return Things{..}
+
+data TrackingId = TrackingId
```
Comment (moleike) [inline src/Line/Bot/Webhook/Events.hs:331]: This could be a `newtype`

File: src/Line/Bot/Webhook/Events.hs [hunk 5]
```diff
@@ -318,3 +327,13 @@ instance FromJSON Things where
...
+   { trackingId      :: Text
+   }
+   deriving (Eq, Show, Generic)
+
+instance FromJSON TrackingId where
+  parseJSON = withObject "videoPlayComplete" $ \o -> do
```
Comment (moleike) [inline src/Line/Bot/Webhook/Events.hs:337]:
```suggestion
  parseJSON = withObject "TrackingId" $ \o -> do
```

File: src/Line/Bot/Webhook/Events.hs [hunk 6]
```diff
@@ -109,6 +110,14 @@ data Event =
                       , timestamp  :: EpochMilli
                       , things     :: Things
                       }
+  | EventUnSend       { source    :: Source
+                      , timestamp :: EpochMilli
+                      }
+  | EventVideoPlayComplete { replyToken :: ReplyToken
+                      , source          :: Source
+                      , timestamp       :: EpochMilli
+                      , trackingId      :: TrackingId
```
Comment (moleike) [inline src/Line/Bot/Webhook/Events.hs:119]:
```suggestion
                      , videoPlayComplete      :: TrackingId
```

**Review by moleike — COMMENTED** [review]
Left some comments.

File: src/Line/Bot/Webhook/Events.hs (hunk 6 above)
Comment (PamornT) [reply src/Line/Bot/Webhook/Events.hs]: Thank for suggest. It Done

File: src/Line/Bot/Webhook/Events.hs (hunk 4 above)
Comment (moleike) [inline src/Line/Bot/Webhook/Events.hs:331]:
```suggestion
newtype TrackingId = TrackingId
```

**Review by moleike — APPROVED** [review]

**Pull request merged**